app/models/user.rb

```diff
@@ -0,0 +1,39 @@
+class User < ActiveRecord::Base
+  has_many :article, dependent: :destroy
+  has_many :comment
+  belongs_to :role
+
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable, :omniauthable, :omniauth_providers => [:facebook]
+
+  before_create :set_default_role
+
+  def is_admin?
+    persisted? and (!role.nil?) and (role.name == "admin") 
```

Comment:
shouldn't have the magic string "admin" here.  Create a constant in your role.rb and call Role::ADMIN here.